src/java/org/apache/cassandra/config/CassandraRelevantProperties.java

```diff
@@ -180,6 +180,9 @@
     /** what class to use for mbean registeration */
     MBEAN_REGISTRATION_CLASS("org.apache.cassandra.mbean_registration_class"),
 
+    /** Which class to use for token metadata provider */
+    CUSTOM_TMD_PROVIDER_PROPERTY("cassandra.custom_token_metadata_provider_class"),
```

Comment:
Do I follow correctly that setting this property is not used anywhere and is not tested? If so, then I guess it will be used in follow up patches or CNDB, but would it be nice to have a test for it? Or is it possible to test?

Reply:
You need to grep for `CUSTOM_TMD_PROVIDER_PROPERTY`:

```
$ gg CUSTOM_TMD
src/java/org/apache/cassandra/config/CassandraRelevantProperties.java:    CUSTOM_TMD_PROVIDER_PROPERTY("cassandra.custom_token_metadata_provider_class"),
src/java/org/apache/cassandra/locator/TokenMetadataProvider.java:import static org.apache.cassandra.config.CassandraRelevantProperties.CUSTOM_TMD_PROVIDER_PROPERTY;
src/java/org/apache/cassandra/locator/TokenMetadataProvider.java:    TokenMetadataProvider instance = CUSTOM_TMD_PROVIDER_PROPERTY.getString() == null ?
src/java/org/apache/cassandra/locator/TokenMetadataProvider.java:                                     CustomTokenMetadataProvider.make(CUSTOM_TMD_PROVIDER_PROPERTY.getString());
```

And the test lives in `CustomTokenMetadataProviderTest.java`

Reply:
I've seen these lines of the code and I don't see that this property is set anywhere. `TokenMetadataProvider` just reads from it, while it was never set anywhere in the code.

`CustomTokenMetadataProviderTest.java` doesn't use the property and doesn't use the interface `TokenMetadataProvider`, which uses the property.

What do I miss?